plugins/ptp_operator/ptp_operator_plugin.go

```diff
@@ -236,10 +245,15 @@ func getCurrentStatOverrideFn() func(e v2.Event, d *channel.DataChan) error {
 					case ptp.PtpClockClassChange:
 						clockClass := fmt.Sprintf("%s/%s", string(ptpInterface), ptpMetrics.ClockClass)
 						data = processDataFn(data, eventManager.GetPTPEventsData(s.SyncState(), s.ClockClass(), clockClass, eventType))
+					case ptp.SyncStateChange:
+						overallSyncState = getOverallState(overallSyncState, s.SyncState())
 					}
 				case ptpMetrics.ClockRealTime:
-					if eventType == ptp.OsClockSyncStateChange {
+					switch eventType {
+					case ptp.OsClockSyncStateChange:
 						data = processDataFn(data, eventManager.GetPTPEventsData(s.SyncState(), s.LastOffset(), string(ptpInterface), eventType))
+					case ptp.SyncStateChange:
+						overallSyncState = getOverallState(overallSyncState, s.SyncState())
```

Comment:
You don't send it again , overall syncState only for Master, and there will be always one Master. So delete this under CLOCK_REALTIME